src/components/Avatar/Avatar.svelte

```diff
@@ -0,0 +1,85 @@
+<script lang="ts">
+    import type { Action } from "svelte/action";
+    
+    // Note: consider `./types.ts`
+    interface AvatarProps {
+        // initials
+        initials?: string;
+        initialsFill?: string;
+        initialsFontSize?: number;
+        initialsBase?: string;
+        initialsRest?: string;
+
+        // image
+        imageBase?:string;
+        imageSrc?: string;
+        imageAlt?: string;
+        imageFallback?: string;
+        imageAction?: Action<HTMLElement, string>;
+        imageActionParams?: string;
+        imageRest?: string;
+
+        // figure
+        figureBase?: string;
+        figureBackground?: string;
+        figureWidth?: string;
+        figureBorder?: string;
+        figureRounded?: string;
+        figureShadow?: string;
+        figureCursor?: string;
+        figureRest?: string;
+    }
+
+    let {
+        // initials
+        initials = "AB",
+        initialsFill = "fill-token",
+        initialsFontSize = 150,
+        initialsBase = "w-full h-full",
+        initialsRest = "",
+
+        //image
+        imageBase = "w-full object-cover",
+        imageSrc = "",
+        imageAlt = "",
+        imageFallback = "",
+        imageAction = () => {},
+        imageActionParams = "",
+        imageRest = "",
+
+        // figure
+        figureBase = "flex aspect-square text-surface-50 font-semibold justify-center items-center overflow-hidden isolate",
+        figureBackground = "bg-surface-400-500-token",
+        figureWidth = "w-16",
+        figureBorder = "",
+        figureRounded = "rounded-full",
+        figureShadow = "",
+        figureCursor = "",
+        figureRest = "",
+    } = $props<AvatarProps>();
+</script>
+
+<figure class="{figureBase} {figureBackground} {figureWidth} {figureBorder} {figureRounded} {figureShadow} {figureCursor} {figureRest}">
+    {#if imageSrc}
+        <img 
+            class="{imageBase} {imageRest}" 
+            src={imageSrc} 
+            alt={imageAlt}
+            use:imageAction={imageActionParams} 
+            onerror={() => imageSrc = imageFallback}
+        />
+    {:else}
+        <svg class="{initialsBase} {initialsRest}" viewBox="0 0 512 512">
```

Comment:
As cool as the SVG approach is for perfectly resizable text. We -should- have access to container queries, clamp, etc in Tailwind now. It might be worth seeing if we can come up with a better approach to this. The SVG has a few quirks - like be slightly off center for reasons I've never been able to pin down!

Again this can be a "TODO" for later though!